src/solve/infer.rs

```diff
@@ -29,7 +28,7 @@ crate struct InferenceSnapshot {
     vars: Vec<InferenceVariable>,
 }
 
-crate type ParameterInferenceVariable = ParameterKind<InferenceVariable>;
+pub(in solve) type ParameterInferenceVariable = ParameterKind<InferenceVariable>;
```

Comment:
I've found that finer grained rules are kind of a pain to maintain and often result in annoying errors from the various privacy rules.

Reply:
Chalk is a big crate, so it seems reasonable to use `pub(in module)`.  I agree that this might be a pain for future refactorings, I personally like making things as private as possible, but no strong opinions.